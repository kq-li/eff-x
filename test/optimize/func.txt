twice : (int -> int) -> int -> int = \f : int -> int. \x : int. {
  x : int = f x;
  x : int = f x;
  return x;
};

add5 : int -> int = add 5;
add10 : int -> int = twice add5;
x : int = add10 2;
_ : unit !output = print x;
