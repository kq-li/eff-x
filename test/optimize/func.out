12
statements executed: 6
